flang/test/Lower/OpenMP/omp-parallel-private-clause.f90

```diff
@@ -0,0 +1,31 @@
+! This test checks lowering of OpenMP parallel Directive with
+! `PRIVATE` clause present.
+
+! RUN: bbc -fopenmp -emit-fir %s -o - | \
+! RUN:   FileCheck %s --check-prefix=FIRDialect
+! RUN: bbc -fopenmp -emit-llvm %s -o - | \
+! RUN:   FileCheck %s --check-prefix=LLVMIRDialect
+
+!FIRDialect: %[[ALPHA:.*]] = fir.alloca i32 {name = "alpha"}
+!FIRDialect: %[[BETA:.*]] = fir.alloca i32 {name = "beta"}
+!FIRDialect: %[[GAMA:.*]] = fir.alloca i32 {name = "gama"}
+!FIRDialect:  omp.parallel private(%[[ALPHA]] : !fir.ref<i32>, %[[BETA]] : !fir.ref<i32>, %[[GAMA]] : !fir.ref<i32>) {
+!FIRDialect:    omp.terminator
+!FIRDialect:  }
+
+!LLVMIRDialect:  %[[ALPHA:.*]] = llvm.alloca %{{.*}} x !llvm.i32 {in_type = i32, name = "alpha"} : (!llvm.i64) -> !llvm.ptr<i32>
+!LLVMIRDialect:  %[[BETA:.*]] = llvm.alloca %{{.*}} x !llvm.i32 {in_type = i32, name = "beta"} : (!llvm.i64) -> !llvm.ptr<i32>
+!LLVMIRDialect:  %[[GAMA:.*]] = llvm.alloca %{{.*}} x !llvm.i32 {in_type = i32, name = "gama"} : (!llvm.i64) -> !llvm.ptr<i32>
+!LLVMIRDialect:  omp.parallel private(%[[ALPHA]] : !llvm.ptr<i32>, %[[BETA]] : !llvm.ptr<i32>, %[[GAMA]] : !llvm.ptr<i32>) {
+!LLVMIRDialect:    omp.terminator
+!LLVMIRDialect:  }
+
+program private_clause
```

Comment:
A few more variants,
-> subroutine arguments.
-> Array

Reply:
Done thanks!
Just wanted clarify that `array` print inside a `omp` region doesn't work(crashes). So test cases captures array, however do not attempt to `print` them.
This may be because of:
https://github.com/flang-compiler/f18-llvm-project/issues/454
I think once we have the minimal set of operations lowered to `fir` dialect, we can start to focus on phase ordering/lower level issues. This will again help us to re-establish end-to-end functional test cases.